tests/causal/cate_learning/test_meta_learners.py

```diff
@@ -394,3 +408,192 @@ def test_causal_s_classification_learner(
     mock_get_unique_treatments.assert_called()
     mock_fit_by_treatment.assert_called()
     mock_simulate_treatment_effect.assert_called()
+
+
+def test_simulate_t_learner_treatment_effect():
+    df = pd.DataFrame({
+        "x1": [1.3, 1.0, 1.8, -0.1],
+        "x2": [10, 4, 15, 6],
+        "treatment": ["A", "B", "A", "control"],
+        "target": [0, 0, 0, 1],
+    })
+
+    treatments = ["A", "B"]
+    control_name = "control"
+    prediction_column = "prediction"
+
+    control_learner = MagicMock()
+    control_learner.side_effect = lambda _: pd.DataFrame({"prediction": [1, 2, 3, 4]})
+
+    def treatment_learner_side_effect(_):
+        return pd.DataFrame({"prediction": [3, 2, 4, 4]})
+
+    treatment_learner = MagicMock()
+    treatment_learner.side_effect = treatment_learner_side_effect
```

Comment:
Why not using a lambda function as well? 😄

Reply:
Nice! :D